lib/puppet_x/puppetlabs/powershell/powershell_manager.rb

```diff
@@ -55,7 +55,7 @@ def initialize(cmd, debug, pipe_timeout = 30)
         # wait for the pipe server to signal ready, and fail if no response in 10 seconds
 
         # wait up to 30 seconds in 0.2 second intervals to be able to open the pipe
-        sleep_interval = 0.2
+        sleep_interval = 0.01
```

Comment:
I'm not sure I understand changing the sleep interval here from 2/10th of a second to 1/100th of a second?

In this loop, we do emit debug output, so this will produce 20x more output while waiting for the pipe to start - which is probably undesirable (even 5x /second is a lot as it stands).

I'm guessing you're doing this as it was proving tricky to get a stalled pipe started up for the sake of tests... I wonder if there's another mechanism we can use to induce that.

Reply:
We were in fact wondering if there was some known reason for picking the original .2 interval value. Yes, we got intermittent passes and fails on the rspec tests with a value as "large" as .2.

Reply:
You're exactly right. That's something we need to investigate before we merge this in - we were getting passing tests even at `0.2` - maybe we make both the timeout and the sleep duration overwritable?

Reply:
I just looked at the code again and stand corrected on emitting debug output - I confused this was some other code... so ignore that part of my comment!

Still wondering about practical implications of short(er) sleeps.

Reply:
Ninja'd by @RandomNoun7. 😛

Reply:
Original 0.2 was fairly random - didn't want to wait for full seconds to elapse while waiting, but didn't want to get into too tight of a sleep / wake loop.  It wasn't chosen very scientifically though, so not sure where we start to get diminishing returns on too short of a sleep time (or even if we do!)

Reply:
I'v done a number of tests at this point and I'm not finding a down side to the shorter test interval. You could argue that we risk spiking the processor with context switches, but we risk doing so really only on our own infrastructure machines, and for an exceedingly short period of time.